src/components/modal/modal.e2e.ts

```diff
@@ -303,83 +303,90 @@ describe("opening and closing behavior", () => {
+          <button id="${button1Id}">Focus1</button>
+          <button id="${button2Id}">Focus2</button>
         </div>
       </calcite-modal>`
     );
+    await skipAnimations(page);
+    await page.waitForChanges();
     const modal = await page.find("calcite-modal");
-    let $button1;
-    let $button2;
-    let $close;
-    await page.$eval(".btn-1", (elm) => ($button1 = elm));
-    await page.$eval(".btn-2", (elm) => ($button2 = elm));
-    await page.$eval("calcite-modal", (elm) => {
-      $close = elm.shadowRoot.querySelector(".close");
-    });
-    await modal.setProperty("open", true);
+    modal.setProperty("open", true);
     await page.waitForChanges();
-    expect(document.activeElement).toEqual($close);
+
+    expect(await isElementFocused(page, `.${CSS.close}`, { shadowed: true })).toBe(true);
     await page.keyboard.press("Tab");
-    expect(document.activeElement).toEqual($button1);
+    expect(await isElementFocused(page, `#${button1Id}`)).toBe(true);
     await page.keyboard.press("Tab");
-    expect(document.activeElement).toEqual($button2);
+    expect(await isElementFocused(page, `#${button2Id}`)).toBe(true);
+
     await page.keyboard.press("Tab");
-    expect(document.activeElement).toEqual($close);
+    expect(await isElementFocused(page, `.${CSS.close}`, { shadowed: true })).toBe(true);
     await page.keyboard.down("Shift");
     await page.keyboard.press("Tab");
-    expect(document.activeElement).toEqual($button2);
+    expect(await isElementFocused(page, `#${button2Id}`)).toBe(true);
+
     await page.keyboard.press("Tab");
-    expect(document.activeElement).toEqual($button1);
+    expect(await isElementFocused(page, `#${button1Id}`)).toBe(true);
   });
 
   it("restores focus to previously focused element when closed", async () => {
+    const initiallyFocusedId = "initially-focused";
+    const initiallyFocusedIdSelector = `#${initiallyFocusedId}`;
     const page = await newE2EPage();
-    await page.setContent(`<button>Focus</button><calcite-modal></calcite-modal>`);
+    await page.setContent(
+      html`
+        <button id="${initiallyFocusedId}">Focus</button>
+        <calcite-modal></calcite-modal>
+      `
+    );
+    await skipAnimations(page);
     const modal = await page.find("calcite-modal");
-    let $button;
-    await page.$eval("button", (elm: any) => {
-      $button = elm;
-      $button.focus();
+    await page.$eval(initiallyFocusedIdSelector, (button: HTMLButtonElement) => {
+      button.focus();
     });
     await modal.setProperty("open", true);
     await page.waitForChanges();
     await modal.setProperty("open", false);
     await page.waitForChanges();
-    expect(document.activeElement).toEqual($button);
+    expect(await isElementFocused(page, initiallyFocusedIdSelector)).toBe(true);
   });
 
   it("traps focus within the modal when open and disabled close button", async () => {
+    const button1Id = "button1";
+    const button2Id = "button2";
```

Comment:
Nitpick: we can probably move these consts outside of the `it` scope at the top of the file so we don't have to declare them more than once

Reply:
I'll do this in a follow-up PR.